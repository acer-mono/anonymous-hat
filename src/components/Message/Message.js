import React from 'react';

class Message extends React.Component {
  render() {
    const { user, message } = this.props;
    return (
      <>
        <div className="card mb-3">
          <div className="card-header font-weight-bold">{user}</div>
          <div className="card-body">{message}</div>
        </div>
      </>
    );
  }
}

export default Message;
