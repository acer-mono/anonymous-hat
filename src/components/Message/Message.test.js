import React from 'react';
import { mount } from 'enzyme';
import Message from './Message';
import renderer from 'react-test-renderer';

describe('Message', () => {
  test('Message shows nick and message', () => {
    const nick = 'test';
    const message = 'test';

    const component = mount(<Message user={nick} message={message} />);
    expect(component.text()).toContain(nick);
    expect(component.text()).toContain(message);
  });

  test('Message test with snapshot', () => {
    const user = 'test';
    const message = 'test';

    const component = renderer.create(<Message user={user} message={message} />);
    let componentSnapshot = component.toJSON();
    expect(componentSnapshot).toMatchSnapshot();
  });
});
