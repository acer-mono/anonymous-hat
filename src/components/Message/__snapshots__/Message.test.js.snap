// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Message test with snapshot 1`] = `
<div
  className="card mb-3"
>
  <div
    className="card-header font-weight-bold"
  >
    test
  </div>
  <div
    className="card-body"
  >
    test
  </div>
</div>
`;
